packages/at_secondary_server/test/keys_verb_test.dart

```diff
@@ -0,0 +1,800 @@
+      inboundConnection.metadata.enrollmentId = enrollId;
+      final enrollJson = {
+        'sessionId': '123',
+        'appName': 'wavi',
+        'deviceName': 'pixel',
+        'namespaces': {'name': 'wavi', 'access': 'rw'},
+        'apkamPublicKey': 'testPublicKeyValue',
+        'requestType': 'newEnrollment',
+        'approval': {'state': 'approved'}
+      };
+      var keyName = '$enrollId.new.enrollments.__manage@alice';
+      await secondaryKeyStore.put(
+          keyName, AtData()..data = jsonEncode(enrollJson));
+
+      var encryptedSelfEncryptionKey =
+          'N0bmvnW1k5oKL+/6X3HresMyG/z6yBmxzgtrn8CMEofWgxJo8RSBXIqvdNj9ZOHO';
+      var valueJson = {};
+      valueJson['value'] = encryptedSelfEncryptionKey;
+      await secondaryKeyStore.put(
+          '$enrollId.$defaultSelfEncryptionKey.$enrollManageNamespace@alice',
+          AtData()..data = jsonEncode(valueJson));
+
+      var keysGetCommand = 'keys:get:self';
+      await keysVerbHandler.process(keysGetCommand, inboundConnection);
+      var keysList = decodeResponseAsList(inboundConnection.lastWrittenData!);
+      expect(keysList, isNotEmpty);
+      expect(keysList[0],
+          '$enrollId.$defaultSelfEncryptionKey.$enrollManageNamespace@alice');
+    });
+
+    test('keys verb invalid syntax - invalid operation', () {
+      var verb = Keys();
+      var command = 'keys:update:hello';
+      var regex = verb.syntax();
+      expect(
+          () => getVerbParam(regex, command),
+          throwsA(predicate((dynamic e) =>
+              e is InvalidSyntaxException && e.message == 'Syntax Exception')));
+    });
+
+    test(
+        'keys verb without specifying the operation invalid syntax - invalid operation',
+        () {
+      var verb = Keys();
+      var command =
+          'keys:public:namespace:__global:keyType:rsa2048:keyName:encryption_1278383933 testPublicKeyValue';
+      var regex = verb.syntax();
+      expect(
+          () => getVerbParam(regex, command),
+          throwsA(predicate((dynamic e) =>
+              e is InvalidSyntaxException && e.message == 'Syntax Exception')));
+    });
+
+    test(
+        'keys verb  - put default  encryption private key and verify keys:get:private',
+        () async {
+      inboundConnection.metadata.isAuthenticated =
+          true; // owner connection, authenticated
+      var enrollId = Uuid().v4();
+      inboundConnection.metadata.enrollmentId = enrollId;
+      final enrollJson = {
+        'sessionId': '123',
+        'appName': 'wavi',
+        'deviceName': 'pixel',
+        'namespaces': {'name': 'wavi', 'access': 'rw'},
+        'apkamPublicKey': 'testPublicKeyValue',
+        'requestType': 'newEnrollment',
+        'approval': {'state': 'approved'}
+      };
+      var keyName = '$enrollId.new.enrollments.__manage@alice';
+      await secondaryKeyStore.put(
+          keyName, AtData()..data = jsonEncode(enrollJson));
+
+      var encryptedDefaultEncryptionPrivateKey =
```

Comment:
Why the hard-coded string?

Reply:
Instead of a dummy value, wanted to test  with actual encrypted value. Server doesn't have dependency for at_demos currently. Is it fine if I add at_demos dependency, get the private key and encrypt with apkam symmetric key?

Reply:
But this is a unit test, nothing to do with at_demos

You can cut keypairs and keys as you need to in your unit tests

Reply:
Done
https://github.com/atsign-foundation/at_server/pull/1496/commits/116fa391e6668f9b4534864e8691d523ce47fb6e